UniversalAlgebra/Structures/Records.lagda

```diff
@@ -36,52 +38,78 @@ open import Relations.Discrete     using ( Arity ; Op ; _|:_ ; _preserves_ )
 open import Relations.Continuous   using ( Rel )
 
 
-ar : Type ℓ₁
-ar = Arity ℓ₀
-
 -- Signatures as records.
-record signature : Type ℓ₁ where
+record signature (𝓞 𝓥 : Level) : Type (lsuc (𝓞 ⊔ 𝓥)) where
  field
-  symbol : Type ℓ₀
-  arity : symbol → ar
+  symbol : Type 𝓞
+  arity : symbol → Type 𝓥
+
+siglev₀ : {𝓞 𝓥 : Level} → signature 𝓞 𝓥 → Level
+siglev₀ {𝓞}{𝓥} _ = 𝓞
+
+siglev₁ : {𝓞 𝓥 : Level} → signature 𝓞 𝓥 → Level
+siglev₁ {𝓞}{𝓥} _ = 𝓥
 
 open signature public
 
+module _ {𝓞₀ 𝓥₀ 𝓞₁ 𝓥₁ : Level} where
 
-record structure
- (𝐹 : signature){α : Level}
- (𝑅 : signature){ρ : Level} : Type (lsuc (α ⊔ ρ)) where
- field
-  carrier : Type α
-  op  : ∀ (𝑓 : symbol 𝐹) → Op  carrier (arity 𝐹 𝑓)      -- interpret. of operations
-  rel : ∀ (𝑟 : symbol 𝑅) → Rel carrier (arity 𝑅 𝑟) {ρ}  -- interpret. of relations
+ record structure (𝐹 : signature 𝓞₀ 𝓥₀)(𝑅 : signature 𝓞₁ 𝓥₁)
+                  {α : Level}{ρ : Level} : Type (𝓞₀ ⊔ 𝓥₀ ⊔ 𝓞₁ ⊔ 𝓥₁ ⊔ (lsuc (α ⊔ ρ))) where
+  field
+   carrier : Type α
+   op  : ∀ (f : symbol 𝐹) → Op  carrier (arity 𝐹 f)      -- interpret. of operations
+   rel : ∀ (r : symbol 𝑅) → Rel carrier (arity 𝑅 r) {ρ}  -- interpret. of relations
+
+ open structure public
+
+
+ module _ {𝐹 : signature 𝓞₀ 𝓥₀}
+          {𝑅 : signature 𝓞₁ 𝓥₁}
+          where
+
+  -- Syntactic sugar for interpretation of operation
+  _ʳ_ : ∀ {α ρ} → (r : symbol 𝑅)(𝒜 : structure 𝐹 𝑅 {α}{ρ}) → Rel (carrier 𝒜) ((arity 𝑅) r) {ρ}
+  r ʳ 𝒜 = λ a → ((rel 𝒜) r) a
```

Comment:
Can't you eta-contract? Which shows that this superscript r is exactly `flip rel`, right?

Reply:
Yes, I suspect you're right.  Nice catch.

Reply:
fixed